tensorflow/core/kernels/parse_tensor_op.cc

```diff
@@ -65,4 +66,33 @@ class ParseTensorOp : public OpKernel {
 
 REGISTER_KERNEL_BUILDER(Name("ParseTensor").Device(DEVICE_CPU), ParseTensorOp);
 
+
+template <typename T>
+class SerializeTensorOp : public OpKernel {
+ public:
+  explicit SerializeTensorOp(OpKernelConstruction* context)
+      : OpKernel(context) {}
+
+  void Compute(OpKernelContext* context) override {
+    const Tensor& tensor = context->input(0);
+    TensorProto proto;
+    if (tensor.dtype() == DT_STRING) {
```

Comment:
There are comments in the codebase where people say:

// Treat DT_STRING specially, so that tensor_util.MakeNdarray in Python can
// convert the TensorProto to string-type numpy array. MakeNdarray does not
// work with strings encoded by AsProtoTensorContent() in tensor_content.

So presumably the same issue is here, but it would be good to validate this is really necessary.